src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -1934,6 +1934,31 @@ public void ChainWith10DataAvailableHeaders_PartialValidationSucceededOnTheFirst
             listOfHeaders.First().ChainedHeader.HashBlock.Should().Be(secondHeader.HashBlock);
         }
 
+        /// Issue 31 @ Chain is 2 blocks long, CT is header 1, call FullValidationRequired on header 2.
+        /// Make sure that reorg that is returned is true.
+        /// </summary>
+        [Fact]
+        public void ChainOfHeaders_CallFullValidationRequiredOnBlockBeyondConsensusTip_ReorgIsRequired()
+        {
+            // Chain header tree setup.
+            const int initialChainSize = 1;
+            TestContext testContext = new TestContextBuilder().WithInitialChain(initialChainSize).Build();
+            ChainedHeaderTree chainedHeaderTree = testContext.ChainedHeaderTree;
+            ChainedHeader initialChainTip = testContext.InitialChainTip;
+
+            const int chainExtension = 1;
+            ChainedHeader chainTip = testContext.ExtendAChain(chainExtension, initialChainTip);
+            List<BlockHeader> listOfChainHeaders = testContext.ChainedHeaderToList(chainTip, 1);
+
+            // Chain is 2 blocks long: h1=h2.
+            chainedHeaderTree.ConnectNewHeaders(1, listOfChainHeaders);
+
+            // Call FullValidationRequired on h2.
+            chainedHeaderTree.PartialValidationSucceeded(chainTip, out bool reorgRequired);
+
```

Comment:
Please add assert that header.BlockValidationState == ValidationState.PartiallyValidated;